hfile_libcurl.c

```diff
@@ -312,6 +365,265 @@ static int add_callback_headers(hFILE_libcurl *fp) {
     return -1;
 }
 
+static int read_auth_json(auth_token *tok, hFILE *auth_fp) {
+    hts_json_token t;
+    kstring_t str = {0, 0, NULL};
+    char *token = NULL, *type = NULL, *expiry = NULL;
+    int ret = 'i';
+
+    if ((ret = hts_json_fnext(auth_fp, &t, &str)) != '{') goto error;
+    while (hts_json_fnext(auth_fp, &t, &str) != '}') {
+        if (t.type != 's') {
+            ret = '?';
+            goto error;
+        }
+        if (strcmp(t.str, "access_token") == 0) {
+            if ((ret = hts_json_fnext(auth_fp, &t, &str)) != 's') goto error;
+            token = ks_release(&str);
+        } else if (strcmp(t.str, "token_type") == 0) {
+            if ((ret = hts_json_fnext(auth_fp, &t, &str)) != 's') goto error;
+            type = ks_release(&str);
+        } else if (strcmp(t.str, "expires_in") == 0) {
+            if ((ret = hts_json_fnext(auth_fp, &t, &str)) != 'n') goto error;
+            expiry = ks_release(&str);
+        } else if (hts_json_fskip_value(auth_fp, '\0') != 'v') {
+            ret = '?';
+            goto error;
+        }
+    }
+
+    if (!token || (type && strcmp(type, "Bearer") != 0)) {
```

Comment:
https://tools.ietf.org/html/rfc7230#section-3.2 states that http headers are case insensitive.  This should probably be another strcasecmp.  Similarly elsewhere.

Reply:
2. Is fixed.
1. Is beyond the scope of this PR.

Reply:
These aren't HTTP headers, they're JSON values.  You need to be looking at [RFC 6749](https://tools.ietf.org/html/rfc6749) and [RFC 6750](https://tools.ietf.org/html/rfc6750).
I can't find anything to say they are supposed to be case-insensitive, although I suppose it wouldn't do any harm to make them.